azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/AzureOperationRef.java

```diff
@@ -6,20 +6,17 @@
 package com.microsoft.azure.toolkit.lib.common.operation;
 
 import com.microsoft.azure.toolkit.lib.common.utils.Utils;
-import lombok.Builder;
+import com.microsoft.azure.toolkit.lib.common.utils.aspect.ExpressionUtils;
+import com.microsoft.azure.toolkit.lib.common.utils.aspect.MethodInvocation;
 import lombok.Getter;
+import lombok.experimental.SuperBuilder;
 
-import java.lang.reflect.Method;
+import java.util.Arrays;
 import java.util.Objects;
 
 @Getter
-@Builder
-public class AzureOperationRef implements IAzureOperation {
-    private final Method method;
-    private final String[] paramNames;
-    private final Object[] paramValues;
-    private final Object instance;
-
+@SuperBuilder
+public class AzureOperationRef extends MethodInvocation implements IAzureOperation {
```

Comment:
Shall we change `MethodInvocation` to interface and make `AzureOperationRef` implement the interface? Is there any case that we need to use `MethodInvocation` rather than `AzureOperationRef`?

Reply:
> Is there any case that we need to use MethodInvocation rather than AzureOperationRef?

yes.